### ISSUE TITLE
en tokenizer model not loading if spacy installed in user home

### ISSUE THREAD

**Issue by nkrot** [explosion/spaCy#1962]
I want to try spacy tokenizer for English. So, I have installed spacy in user space with

    > pip3 install spacy --user

that leaves me with

    > python3 -m spacy validate

    Installed models (spaCy v2.0.7)
    $HOME/.local/lib/python3.6/site-packages/spacy

My guess is that the language-specific models are here: `$HOME/.local/lib/python3.6/site-packages/spacy/lang/en`. This directory is not empty, so i guess `en` models have been installed, at least i see there a file named `tokenizer_exceptions.py`.

However, when using spacy in a script, i receive an error:
```
>>> import spacy
>>> nlp = spacy.load("en")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "$HOME/.local/lib/python3.6/site-packages/spacy/__init__.py", line 19, in load
    return util.load_model(name, **overrides)
  File "$HOME/.local/lib/python3.6/site-packages/spacy/util.py", line 119, in load_model
    raise IOError("Can't find model '%s'" % name)
OSError: Can't find model 'en'
```

If I try to install language models like follows:

    python3 -m spacy download en

the installer tries to put data into system-wide location, which is not desired

    error: could not create '/usr/local/lib/python3.6/site-packages/en_core_web_sm': Permission denied

Basically, two questions: Were the language models installed? And if not, how to install them in the user home under `.local/` where `pip` installs stuff?

## Info about spaCy

* **spaCy version:** 2.0.7
* **Platform:** Linux-3.10.0-327.22.2.el7.x86_64-x86_64-with-centos-7.2.1511-Core
* **Python version:** 3.6.4

Thanks in advance,

**Comment by ines** [comment]
spaCy models are Python packages and when you run the `spacy download` command, the respective URL to the pip-installable file is fetched from our [models repo](https://github.com/explosion/spacy-models) and downloaded via pip in a subprocess. I think you might be experiencing the same issue as #1220 (also see #1456), since the `--user` flag isn't set in the subprocess run by spaCy.

The easiest workaround for now would be to [install the model manually](https://spacy.io/usage/models#download-pip). For example, [here's the download of the small English model `en_core_web_sm`](https://github.com/explosion/spacy-models/releases/tag/en_core_web_sm-2.0.0) (which is the default model that's also downloaded via `spacy download en`). The attached `.tar.gz` archive is an installable package, so you should be able to run:

```bash
pip install /path/to/en_core_web_sm-2.0.0.tar.gz --user
```

This will let you load the model via `spacy.load('en_core_web_sm')`. [Here's some more info](https://spacy.io/usage/models#usage) on how model loading works.

The content in `spacy.lang.en` is the English *language data*, i.e. the tokenizer exceptions, punctuation rules, lemmatizer data etc. All spaCy models specify a language code (e.g. `"en"`) that tells spaCy which language data to initialise. So when you load a model, spaCy will create the language class and processing pipeline, and load in the binary weights so you can use the features that require a statistical model – like the tagger, parser or entity recognizer. [Here's](https://spacy.io/usage/processing-pipelines#pipelines) a more detailed explanation of this.

If you only want to use the tokenizer, you don't even have to download and load a model – instead, you can simply import and use the respective language class:

```python
from spacy.lang.en import English

nlp = English()
doc = nlp(u"This text will be tokenized but won't have POS tags, dependencies or entities")
```

**Comment by nkrot** [comment]
Thank you, @ines. The instructions helped.

For the record, I did the following

    > wget https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-2.0.0/en_core_web_sm-2.0.0.tar.gz
    > pip3 install --user path/to/en_core_web_sm-2.0.0.tar.gz
    > python3 -m spacy link en_core_web_sm en

The result are the following objects that appeared on the filesystem

    # installed by pip3
    $HOME/.local/lib/python3.6/site-packages/en_core_web_sm/
    # symbolic link created by spacy link command
    $HOME/.local/lib/python3.6/site-packages/spacy/data/en/

Now I can load language model by specifying the language code

    >>> import spacy
    >>> nlp = spacy.load("en")

I exactly need the latter ability, because I use `NeuroNER` and do not want to mess up the code of the latter by changing the `nlp.load()`.

Thank you very much for you help.
